server/src/test/java/io/unitycatalog/server/base/BaseCRUDTestWithMockCredentials.java

```diff
@@ -0,0 +1,154 @@
+            .sessionToken("test-session-token")
+            .build();
+    serverProperties.entrySet().stream()
+        .filter(e -> e.getKey().toString().contains("s3.bucketPath"))
+        .forEach(
+            e ->
+                doReturn(credentials)
+                    .when(awsCredentialVendor)
+                    .vendAwsCredentials(
+                        argThat(
+                            arg ->
+                                arg.getStorageScheme().equals("s3")
+                                    && arg.getStorageBase().contains(e.getValue().toString()))));
+
+    azureCredentialVendor = mock(AzureCredentialVendor.class);
+    AzureCredential azureCredential =
+        AzureCredential.builder()
+            .sasToken("test-sas-token")
+            .expirationTimeInEpochMillis(System.currentTimeMillis() + 6000)
+            .build();
+    serverProperties.entrySet().stream()
+        .filter(e -> e.getKey().toString().contains("adls.storageAccountName"))
+        .forEach(
+            e ->
+                doReturn(azureCredential)
+                    .when(azureCredentialVendor)
+                    .vendAzureCredential(
+                        argThat(
+                            arg ->
+                                arg.getStorageScheme().equals("abfs")
+                                    && arg.getStorageBase().contains(e.getValue().toString()))));
+
+    gcpCredentialVendor = mock(GcpCredentialVendor.class);
+    AccessToken accessToken =
+        new AccessToken("test-token", Date.from(Instant.now().plusSeconds(10 * 60)));
+    serverProperties.entrySet().stream()
+        .filter(e -> e.getKey().toString().contains("gcs.bucketPath"))
+        .forEach(
+            e ->
+                doReturn(accessToken)
+                    .when(gcpCredentialVendor)
+                    .vendGcpToken(
+                        argThat(
+                            arg ->
+                                arg.getStorageScheme().equals("gs")
+                                    && arg.getStorageBase().contains(e.getValue().toString()))));
+
+    credentialOperations =
+        new CredentialOperations(awsCredentialVendor, azureCredentialVendor, gcpCredentialVendor);
+  }
+
+  protected String getTestCloudPath(String scheme, boolean isConfiguredPath) {
+    // test-bucket0 is configured in the properties
+    String bucket = isConfiguredPath ? "test-bucket0" : "test-bucket1";
+    return switch (scheme) {
+        case "s3" -> "s3://" + bucket + "/test";
+        case "abfs", "abfss" -> "abfs://test-container@" + bucket + ".dfs.core.windows.net/test";
+        case "gs" -> "gs://" + bucket + "/test";
+        default -> throw new IllegalArgumentException("Invalid scheme");
+    };
+  }
+
+  protected void assertTemporaryCredentials(
+      TemporaryCredentials temporaryCredentials, String scheme) {
+    switch (scheme) {
+      case "s3":
+        assertThat(temporaryCredentials.getAwsTempCredentials()).isNotNull();
```

Comment:
nit:  please reduce code duplication. why are we calling `temporaryCredentials.getAwsTempCredentials()` 4 times?